default/python/AI/colonization/calculate_production.py

```diff
@@ -0,0 +1,203 @@
+import freeOrionAIInterface as fo
+from typing import Iterable, NamedTuple
+
+import AIDependencies
+from buildings import BuildingType
+from freeorion_tools import get_named_real, get_species_tag_value, tech_is_complete
+from turn_state import have_honeycomb
+
+
+def calculate_production(
+    planet: fo.planet, species: fo.species, max_population: float, stability: float, research_list: Iterable
+) -> float:
+    """
+    Calculate how much PP the planet's population could generate with industry focus.
+    This only considers values that actually rely on industry focus, those that do not are handled by
+    calculate_planet_colonization_rating._rate_focus_independent.
+    """
+    if stability <= 0.0:
+        return 0.0
+
+    bonus_modified = _get_production_bonus_modified(planet, stability, research_list)
+    skill_multiplier = get_species_tag_value(species.name, AIDependencies.Tags.INDUSTRY)
+    bonus_by_policy = _get_production_bonus_mod_by_policy(stability)
+    policy_multiplier = _get_policy_multiplier(stability)
+    bonus_unmodified = _get_production_bonus_unmodified(planet, stability)
+    bonus_flat = _get_production_flat(planet, stability, research_list)
+    per_population = (
+        (AIDependencies.INDUSTRY_PER_POP + bonus_modified) * skill_multiplier + bonus_by_policy
+    ) * policy_multiplier + bonus_unmodified
+    return max_population * per_population + bonus_flat
+
+
+class _ProductionBonus(NamedTuple):
+    available: bool
+    min_stability: float
+    value: float
+
+
+def _get_production_bonus_modified(planet: fo.planet, stability: float, research_list: Iterable):
+    """
+    Calculate bonus production per population which is added before multiplication with the species skill value.
```

Comment:
I'd say it's not calculating a real bonus, but it predicts future bonuses.  Let's reflect this in the docs.